Libraries/hipSOLVER/sygvj/README.md

```diff
@@ -0,0 +1,107 @@
+    - `int lda`: Leading dimension of matrix $A$.
+    - `double *B`: Pointer to matrix $B$ in device memory.
+    - `int ldb`: Leading dimension of matrix $B$.
+    - `double *D`: Pointer to array $W$, where the resulting eigenvalues are written.
+    - `double *work`: Pointer to working space in device memory.
+    - `int lwork`: Size of the working space.
+    - `int *devInfo`: Convergence result of the function in device memory. If 0, the algorithm converged, if greater than 0 and less or equal to `n` then `devInfo`-th leading minor of `B` is not positive definite, if equal to `n+1` than convergence is not achieved. Also, for CUDA backend, if `devInfo = -i` for $0 < i \leq n$, then the the $i^{th}$ parameter is wrong (not counting the handle).
+    - `hipsolverSyevjInfo_t params`: Pointer to the structure with parameters of solver, that should be created by function `hipsolverDnCreateSyevjInfo(&params)`. Solver has two parameters:
+        - Tolerance `tol`, set by function `hipsolverDnXsyevjSetTolerance(syevj_params, tol)`, default value of tolerance is machine zero.
+        - Maximal number of sweeps to obtain convergence `max_sweeps`, set by function `hipsolverDnXsyevjSetMaxSweeps(syevj_params, max_sweeps)`, default value is 100.
+
+    Return type: `hipsolverStatus_t`.
+- `hipsolver[SD]sygvj_bufferSize` allows to obtain the size (in bytes) needed for the working space for the `hipsolver[SD]sygvj` function. The character matched in `[SD]` coincides with the one in `hipsolver[SD]sygvj`.
+
+    This function accepts the following input parameters:
+    - `hipsolverHandle_t handle`
+    - `hipsolverEigType_t itype`: Specifies the type of eigensystem problem.
+    - `hipsolverEigMode_t jobz`: Specifies whether the eigenvectors should also be calculated besides the eigenvalues.
+    - `hipSolverFillMode_t uplo`: Specifies whether the upper or lower triangle of the symmetric matrix is stored.
+    - `int n`: Number of rows and columns of $A$.
+    - `double *A`: Pointer to matrix $A$ in device memory.
+    - `int lda`: Leading dimension of matrix $A$.
+    - `double *B`: Pointer to matrix $B$ in device memory.
+    - `int ldb`: Leading dimension of matrix $B$.
+    - `double *D`: Pointer to array $W$ in device memory, where the resulting eigenvalues are written.
+    - `int *lwork`: The required buffer size is written to this location.
+    - `hipsolverSyevjInfo_t params`: Pointer to the structure with parameters of solver.
+
+    Return type: `hipsolverStatus_t`.
+
+## Used API surface
+### hipSOLVER
+Types:
+- `hipsolverHandle_t`
+- `hipsolverSyevjInfo_t`
+- `hipsolverEigType_t`
+- `hipsolverEigMode_t`
+- `hipsolverFillMode_t`
+
+Functions:
+- `hipsolverCreate`
+- `hipsolverDestroy`
+- `hipsolverDnCreateSyevjInfo`
+- `hipsolverDnDestroySyevjInfo`
+- `hipsolverDnXsyevjSetTolerance`
+- `hipsolverDnXsyevjSetMaxSweeps`
+- `hipsolverDnXsyevjGetResidual`
+- `hipsolverDnXsyevjGetSweeps`
+- `hipsolverDsygvj_bufferSize`
+- `hipsolverDsygvj`
```

Comment:
It looks like you're mixing calls to the compatibility API with calls to the regular API, which is perfectly valid but it would be nice to be consistent.

Reply:
Oopsie, we shouldn't be mixing these. Should be fixed now.